src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -59,8 +62,12 @@ public HttpResponseMessage Invoke(string name, [FromBody] FunctionInvocation inv
         [Route("admin/functions/{name}/status")]
         public FunctionStatus GetFunctionStatus(string name)
         {
-            FunctionStatus status = new FunctionStatus();
             Collection<string> functionErrors = null;
+            FunctionDescriptor function = _scriptHostManager.Instance.Functions.FirstOrDefault(p => p.Name.ToLowerInvariant() == name.ToLowerInvariant());
+            FunctionStatus status = new FunctionStatus
+            {
+                Metadata = function?.Metadata
```

Comment:
so the problem with getting that from `WebScriptHostManager` is that we can't new up that object. The only way we know in the cli how to get a hold of that object is by actually instantiating the host itself, then grab it from the dependency resolver we get back.